Feature/restaurants page

keren-elbaz-m opened the pull request.

keren-elbaz-m committed:
Filters in restaurant page

Initial layout for restaurants filtering – added tab and dropdown components and connected them to the page

keren-elbaz-m committed:
Restaurant cards

Implemented responsive grid layout for restaurant page cards

keren-elbaz-m committed:
filter cards

render cards based on active tab selection

keren-elbaz-m committed:
navbar highlight active menu

highlight active menu item based on pathname

DanielBaronMoveo commented on src/app/api/restaurants/route.ts:
```diff
@@ -1,6 +1,29 @@
+  let filtered = restaurants;
+
+  switch (filter) {
+    case "popular":
```
Let's create a proper enum for the diverse filters and use it instead of the hardcoded strings within the logic.

```
export enum RestaurantFilter {
  ALL = "all",
  POPULAR = "popular",
  NEW = "new",
  OPEN = "open",
}

keren-elbaz-m replied:
done

DanielBaronMoveo commented on src/app/api/restaurants/route.ts:
```diff
@@ -1,6 +1,29 @@
 
-export async function GET() {
-  return NextResponse.json(restaurants);
+export async function GET(request: Request) {
```
We are missing try/catch blocks within the async scope.

keren-elbaz-m replied:
done

DanielBaronMoveo commented on src/app/api/restaurants/route.ts:
```diff
@@ -1,6 +1,29 @@
-  return NextResponse.json(restaurants);
+export async function GET(request: Request) {
+  const url = new URL(request.url);
+  const filter = url.searchParams.get("filter");
```
Default to 'all' if no filter provided
 

    const filter = filterParam || RestaurantFilter.ALL;

keren-elbaz-m replied:
done :)

DanielBaronMoveo commented on src/app/api/restaurants/route.ts:
```diff
@@ -1,6 +1,29 @@
+
+  let filtered = restaurants;
+
+  switch (filter) {
```
We can give up on the switch here and turn it into a cool logic with something like:

```
const filterStrategies: Record<RestaurantFilter, (restaurants: Restaurant[]) => Restaurant[]> = {
  [RestaurantFilter.ALL]: (restaurants) => restaurants,
  [RestaurantFilter.POPULAR]: (restaurants) => restaurants.filter((r) => r.isPopular),
  [RestaurantFilter.NEW]: (restaurants) => restaurants.filter((r) => r.isNew),
  [RestaurantFilter.OPEN]: (restaurants) => restaurants.filter((r) => r.isOpen),
};
```

Afterwards just use it within the scope with the correct filter selected.
```
const filteredRestaurants = filterStrategies[filter](restaurants);
```

keren-elbaz-m replied:
done :)

DanielBaronMoveo commented on src/components/Filter/Filter.module.scss:
```diff
@@ -0,0 +1,95 @@
+//   @media screen and (min-width: 1024px) {
+//     justify-content: center;
+//     padding: 32px 0;
+//   }
```
Remove.

keren-elbaz-m replied:
done

DanielBaronMoveo commented on src/components/Filter/Filter.tsx:
```diff
@@ -0,0 +1,42 @@
+
+
+    useEffect(() => {
+        const filter = activeTab === "All" ? "all" : activeTab.toLowerCase().replace(/\s/g, "");
```
Use the enum value for all instead of hardcoded string.

keren-elbaz-m replied:
done

DanielBaronMoveo reviewed (changes requested):
Overall, it looks great. Did a quick sweep over it and will go over it again tomorrow.

Good job! There are though some minor changes I've requested, we'll discuss it tomorrow if something is not clear enough :)

keren-elbaz-m committed:
Update index.ts

Added RestaurantFilter enum

keren-elbaz-m committed:
Update route.ts

Added error handling to restaurants API route using try/catch

keren-elbaz-m committed:
Update route.ts

Refactored restaurant filtering logic using strategy map and made default to 'all' if no filter provided

keren-elbaz-m committed:
Update Filter.module.scss

Removed comment

keren-elbaz-m committed:
Refactored tab filtering logic to use enum-based mapping

Defined `TabLabel` type and `TabLabelToFilterMap` record to map tab labels to filter values and updated `Filter` component to use `TabLabel` and derive API filter from enum mapping

keren-elbaz-m committed:
Merge branch 'dev' into feature/restaurants-page
